### ISSUE TITLE
custom dynamic authorization URL parameters

### ISSUE THREAD

**Issue by djbulsink** [travisghansen/external-auth-server#114]
hi,

In the oidc plugin it's possible to set custom static authorization URL parameters. We would like to have a dynamic option that an specific URL parameters from the parent request are send to the authorization endpoint. That would help us to implement the use case where an end-user is able to suggest a specific idp in keycloak (https://www.keycloak.org/docs/latest/server_admin/#_client_suggested_idp).

Do you think this would be a feature that adds value to EAS? If you like to I can create a MR for this new feature.

**Comment by travisghansen** [comment]
I've implemented this in `next` (a bit different than your PR you submitted however). If you could test out the new images that would be great. The parameters are exactly the same as before but the values can be written in handlebars syntax. You have access to the `req` object as well as the `parentReqInfo` object.

```
...
"custom_authorization_parameters": {
  "some_param": "{{ parentReqInfo.parsedQuery.<param> }}"
}
...
```

**Comment by djbulsink** [comment]
That's a great solution! I'll test it later today.

**Comment by djbulsink** [comment]
I've tested it. Works like a charm!

**Comment by travisghansen** [comment]
Great! I’ll be merging into master/latest in the next week or so with a new tag as well.

**Comment by travisghansen** [comment]
This has been merged to master/latest and in available in `v0.11.0`. Thanks for the idea and PR you submitted to get it there!
